### PR TITLE
PLF-5064 Stop ExecutorService cleanly

### PR DESCRIPTION
**Pull request by boubaker**

### CODE REVIEW

File: exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java [hunk 1]
```diff
@@ -35,6 +37,7 @@
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.Executor;
```
Comment (thomasdelhomenie) [inline exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java]: Useless import now.

File: exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java [hunk 2]
```diff
@@ -335,4 +338,13 @@ public Void run()
...
    }
+
+  @Override
+  public void start() {
+  }
+
+  @Override
+  public void stop() {
+    executor.shutdown();
+  }
```
Comment (thomasdelhomenie) [inline exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java]: The change is ok for me, but I am wondering something: if we want to have a clean stop, we have to make all the services using executors as Startable, which means more and more services will become Startable. Couldn't we find a better way to handle this ? Maybe with a service which handles it in a centralized manner (and will stop cleanly all executors) ? Is it really a problem to have more and more Startable services ?

Comment (thomasdelhomenie) [reply exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java]: By reviewing the Social PR, I just found we actually have such a service : ExecutorServiceManagerImpl :)
Shouldn't we use it (looks like it needs some improvements though) each time we want to use an executor ?

Comment (boubaker) [reply exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java]: Yep, I saw it too, but to be treated as Technical improvement for the future PRs ;-)

File: exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java (hunk 1 above)
Comment (boubaker) [reply exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java]: deleted

File: exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java (hunk 2 above)
Comment (thomasdelhomenie) [reply exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java]: ok, sounds good (I let you open the jira issue ?) :+1:

Comment (boubaker) [reply exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java]: It's your idea, I let describe what's expected ;-)

Comment (thomasdelhomenie) [reply exo.kernel.component.common/src/main/java/org/exoplatform/services/listener/ListenerService.java]: ok

**Pull request merged**